snapcraft/config.py

```diff
@@ -66,6 +70,18 @@ def is_empty(self):
         return True
 
     def load(self):
+        # Local configurations (per project) are supposed to be static.
+        # That's why it's only checked for 'loading' and never written to.
+        # Essentially, all authentication-related changes, like login/logout
+        # or macaroon-refresh, will not be persisted for the next runs.
+        local_config = '.snapcraft.cfg'
+        if os.path.exists(local_config):
+            self.parser.read(local_config)
+            logger.warn(
+                'Using local configuration (`.snapcraft.cfg`), changes will '
+                'not be persisted.')
```

Comment:
Tests added, along with a major refactor for isolating "integrations" implementations.